### PR TITLE
Support saving/serializing empty values

### PR DESCRIPTION
**Pull request by arunagw**
Currently if we want to nullify a attribute on server it's not possible
to do as we reject keys where values are `nil` and `""`.
We want to support this as API accepts sending `nil` and empty values
so that attributes can be nullified.

This PR introduces two new methods to send all attributes to server
event the values are `nil` and empty. These new methods are called
`save_all_attributes` and `update_all_attributes`.

This change was also requested in #221

Changes:
- Update `Nylas::Model` to add new public methods.
- Update `Nylas::Model::Attributes` to add method to serialize all
  attributes.

### CODE REVIEW

File: spec/nylas/model_spec.rb [hunk 1]
```diff
@@ -44,6 +44,93 @@ def example_instance_hash
...
+
+    it "raises a NotImplementedError exception if the model is flagged as read only" do
+      instance = NotUpdatableModel.from_json(example_instance_json, api: api)
+      expect { instance.update_all_attributes(name: "other") }.to raise_error(Nylas::ModelNotUpdatableError)
+    end
+
+    it "raises a MissingFieldError if attempting to set a field that does not exist" do
+      expected_message = "fake_attribute is not a valid attribute for FullModel"
+      instance = FullModel.new
+      expect do
```
Comment (adarsh) [inline spec/nylas/model_spec.rb:128]: Minor: newline after here to separate test phases.

**Review by adarsh — COMMENTED** [review]
LGTM, this seems like an important update!

File: lib/nylas/model.rb [hunk 2]
```diff
@@ -69,6 +69,37 @@ def update(**data)
       raise ModelMissingFieldError.new(e.key, self)
     end
 
+    def save_all_attributes
+      result = if persisted?
+                 raise ModelNotUpdatableError, self unless updatable?
```
[the remark this answers is not in the document]
Comment (arunagw) [reply lib/nylas/model.rb:74]: Yes. It's here https://github.com/nylas/nylas-ruby/blob/main/lib/nylas/errors.rb#L14

File: spec/nylas/model_spec.rb [hunk 3]
```diff
@@ -44,6 +44,93 @@ def example_instance_hash
...
+        payload: update_params.to_json,
+        path: "/collection/1234",
+        query: {}
+      )
+      expect(instance.location).to eq(nil)
+      expect(instance.string).to eq("")
+    end
+
+    it "raises a NotImplementedError exception if the model is flagged as read only" do
+      instance = NotUpdatableModel.from_json(example_instance_json, api: api)
```
[the remark this answers is not in the document]
Comment (arunagw) [reply spec/nylas/model_spec.rb:123]: Done here f8a3f5c74b192dc1a5bb1975d2731fb84d3e47e5

File: spec/nylas/model_spec.rb (hunk 1 above)
Comment (arunagw) [reply spec/nylas/model_spec.rb:132]: Done here f8a3f5c74b192dc1a5bb1975d2731fb84d3e47e5

**Pull request merged**